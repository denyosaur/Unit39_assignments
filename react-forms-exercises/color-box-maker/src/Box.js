import React from "react";

const Box = ({ height, width, backgroundColor, deleteBox }) => {
    const boxStyle = {
        backgroundColor: backgroundColor,
        height: `${height}px`,
        width: `${width}px`
    }

    return <div style={boxStyle}>
        <button onClick={deleteBox}>X</button>
    </div >
}

export default Box;