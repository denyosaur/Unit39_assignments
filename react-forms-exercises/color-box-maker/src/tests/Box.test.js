import { render, fireEvent } from '@testing-library/react';
import Box from '../Box';

//smoke test for Box
test('renders Box without crashing', () => {
    render(<Box height="100" width="100" backgroundColor="black" />);
});

//snapshot test for Box
test('should match snapshot', () => {
    const { asFragment } = render(<Box height="100" width="100" backgroundColor="black" />);
    expect(asFragment()).toMatchSnapshot();
});

//render and deleting box
test("should render and delete Box", () => {
    const { getByText } = render(<Box height="100" width="100" backgroundColor="black" />);
    const btn = getByText("X");
    expect(btn).toBeInTheDocument;

    fireEvent.click(btn);

    expect(btn).not.toBeInTheDocument;
})