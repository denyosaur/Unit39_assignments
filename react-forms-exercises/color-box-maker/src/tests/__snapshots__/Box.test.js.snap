// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    style="background-color: black; height: 100px; width: 100px;"
  >
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
